lib/portage/util/_async/ForkProcess.py

```diff
@@ -29,6 +30,41 @@ class ForkProcess(SpawnProcess):
     # (after the sentinel has become ready).
     _proc_join_interval = 0.1
 
+    def _start(self):
+        if self.fd_pipes or self.logfile:
```

Comment:
I wish we could write this first part (up to retval = ...) in some way which feels less repetitive, but I think the current way is easiest to read, so lgtm